### PR TITLE
K8SPG-375: Merge extension installer image into operator

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPG-375](https://jira.percona.com/browse/K8SPG-375) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
We don't want to maintain a separate image just to install extensions.

**Solution:**
We'll use operator image for the job.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PG version?
- [x] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/users.log) |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/eb306ee3e/custom-extensions.log) |
| We run 11 out of 11|

commit: https://github.com/percona/percona-postgresql-operator/pull/573/commits/eb306ee3eacb031785e53ad89afa323fe582929f
image: `perconalab/percona-postgresql-operator:PR-573-eb306ee3e`

**Review by hors — APPROVED** [review]

File: percona/controller/pgcluster/extensions.go [hunk 1]
```diff
@@ -50,6 +50,13 @@ func ExtensionInstallerContainer(postgresVersion int, spec *v2.ExtensionsSpec, e
 		Image:           spec.Image,
 		ImagePullPolicy: spec.ImagePullPolicy,
 		Command:         []string{"/usr/local/bin/install-extensions.sh"},
+		SecurityContext: &corev1.SecurityContext{
+			RunAsUser: func() *int64 {
+				// postgres user
+				var uid int64 = 26
```
Comment (hors) [inline percona/controller/pgcluster/extensions.go:56]: @egegunes I hope it will work with OpenShift

Comment (egegunes) [reply percona/controller/pgcluster/extensions.go:56]: I think it'll. Our postgres image is also running with this user id

**Review by nmarukovich — DISMISSED** [review]

**Review by pooknull — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/users.log) |
| custom-extensions | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/7e419a5cb/custom-extensions.log) |
| We run 11 out of 11|

commit: https://github.com/percona/percona-postgresql-operator/pull/573/commits/7e419a5cb5a28524d1ca6b385235b5f7ecb242e2
image: `perconalab/percona-postgresql-operator:PR-573-7e419a5cb`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/scheduled-backup.log) |
| self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/users.log) |
| custom-extensions | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-573/9ac6d3cd0/custom-extensions.log) |
| We run 11 out of 11|

commit: https://github.com/percona/percona-postgresql-operator/pull/573/commits/9ac6d3cd0df3737b61cd0f5455071887d370235f
image: `perconalab/percona-postgresql-operator:PR-573-9ac6d3cd0`

**Review by hors — APPROVED** [review]

**Review by nmarukovich — APPROVED** [review]

**Pull request merged**